### PR TITLE
Fix: `None` in "wait" Methods Causing Incorrectly Passing Tests

### LINKED ISSUE #2818: [BUG] Dash Testing: `wait_for_text_to_equal` may incorrectly succeed when used with text `"None"`
**Describe your context**

-  replace the result of `pip list | grep dash` below
```
dash                            2.16.1         
dash-core-components            2.0.0
dash-dangerously-set-inner-html 0.0.2
dash-flow-example               0.0.5
dash-html-components            2.0.0
dash-table                      5.0.0
dash-testing-stub               0.0.2
```

**Describe the bug**

When `wait_for_text_to_equal` is used to wait for the text `"None"`, the function will often succeed even when you would reasonably expect it to fail. 

I think this is part of the reason why the regression in #2733 wasn't caught by the tests.

This behavior is demonstrated by the following test case:

```python
import dash
from dash import html


def test_wait_for_text_to_equal_none(dash_duo):
    app = dash.Dash(__name__)
    app.layout = html.Div(id="my-div", children="Hello world")
    dash_duo.start_server(app)
    dash_duo.wait_for_text_to_equal("#my-div", "None", timeout=4)
```

**Expected behavior**

The test should fail because the contents of the `#my-div` div are never equal to `None` or `"None"`.

**Actual behavior**

The test passes.

**Explanation**

This happens because `wait_for_text_to_equal` checks not only the text content of the element, but also the value of the `value` attribute. ([see here](https://github.com/plotly/dash/blob/f7f8fb4c5893506e35cdeaec141310a95fe1486a/dash/testing/wait.py#L110C13-L113C14)).

If `value` is not defined we get a value of `None`, which is then converted to a string and therefore matches the string `"None"`.

So `dash_duo.wait_for_text_to_equal("#my-div", "None")` _always_ succeeds unless the target element has a defined `value`.

**Proposed solutions**

IMO the cleanest solution would be to modify `wait_for_text_to_equal` to check _only_ the element's text, and add a new function `wait_for_value_to_equal` which checks the value (or a generalized `wait_for_attr_to_equal` function). This would break backwards compatibility.

Alternatively we could have `wait_for_text_to_equal` ignore `value` if value is not defined, or issue a warning when used with the text `"None"`.

### PR DESCRIPTION
**Pull request by TillerBurr**
Closes #2818. This is my take on solving the issue. This method ignores the value prop if it is `None`. I think there is merit to adding a `wait_for_value_to_equal`, but this solves the immediate problem, while causing as little friction as possible.

As mentioned in #2818, there are also potential issues in the other methods in the `dash.testing.wait` namespace. Not sure if those cases need to be handled here.

## Contributor Checklist

- [x] I have broken down my PR scope into the following TODO tasks
   -  [x] Check if the value attribute is None and don't compare if it is.
   -  [x] Add tests to make sure the `None` isn't cast to a string.
- [x] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))
- [x] I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR

### optionals

- [ ] I have added entry in the `CHANGELOG.md`
- [ ] If this PR needs a follow-up in **dash docs**, **community thread**, I have mentioned the relevant URLS as follows
    -  [ ] this GitHub [#PR number]() updates the dash docs
    -  [ ] here is the show and tell thread in Plotly Dash community

### CODE REVIEW

**Comment by TillerBurr** [comment]
Tests are failing due to bad test that is fixed in https://github.com/plotly/dash/pull/2816. Once that PR is merged and this merged into this PR, the tests will all pass.

File: tests/integration/test_duo.py [hunk 1]
```diff
@@ -27,10 +32,33 @@ def test_duo001_wait_for_text_error(dash_duo):
         == "text -> invalid not found inside element within 1.0s, found: Content"
     )
 
+    with pytest.raises(TimeoutException) as err:
+        dash_duo.wait_for_contains_text("#content", "None", timeout=1.0)
+
+    assert (
```
Comment (emilykl) [inline tests/integration/test_duo.py:38]: 👍

File: dash/testing/wait.py [hunk 2]
```diff
@@ -107,10 +107,9 @@ def __call__(self, driver):
         try:
             elem = self._get_element(driver)
             logger.debug("text to equal {%s} => expected %s", elem.text, self.text)
-            return (
-                str(elem.text) == self.text
-                or str(elem.get_attribute("value")) == self.text
-            )
+            if (value := elem.get_attribute("value")) is not None:
+                return str(elem.text) == self.text or str(value) == self.text
+            return str(elem.text) == self.text
```
Comment (emilykl) [inline dash/testing/wait.py:112]:
```suggestion
            value = elem.get_attribute("value")
            return str(elem.text) == self.text or (value is not None and str(value) == self.text)
```

IMO better to avoid the walrus operator here as it's clearer without. Same goes for the other function.

**Comment by emilykl** [comment]
I like this solution. Need to think through possible consequences though as this is technically a breaking change in Dash Testing.

File: dash/testing/wait.py [hunk 3]
```diff
@@ -62,9 +63,9 @@ def __call__(self, driver):
         try:
             elem = driver.find_element(By.CSS_SELECTOR, self.selector)
             logger.debug("contains text {%s} => expected %s", elem.text, self.text)
-            return self.text in str(elem.text) or self.text in str(
-                elem.get_attribute("value")
-            )
+            if (value := elem.get_attribute("value")) is not None:
```
Comment (emilykl) [inline dash/testing/wait.py:66]: @TillerBurr Can you change the logic here as well to remove the walrus operator? Once that's done, this PR looks good to me.

Comment (TillerBurr) [reply dash/testing/wait.py]: Oops. That one is updated now.

**Review by T4rk1n — APPROVED** [review]
:woman_dancing:

File: dash/testing/wait.py [hunk 4]
```diff
@@ -62,8 +63,9 @@ def __call__(self, driver):
         try:
             elem = driver.find_element(By.CSS_SELECTOR, self.selector)
             logger.debug("contains text {%s} => expected %s", elem.text, self.text)
-            return self.text in str(elem.text) or self.text in str(
-                elem.get_attribute("value")
+            value = elem.get_attribute("value")
+            return str(elem.text) in self.text or (
```
Comment (emilykl) [inline dash/testing/wait.py:67]: This is swapped -- should be `self.text in str(elem.text)` and `self.text in str(value)`

Comment (TillerBurr) [reply dash/testing/wait.py]: Thanks. Got confused on the order. Fixed.

**Review by emilykl — APPROVED** [review]
Looks good to me @TillerBurr -- thanks for the contribution! ✨ 

There's some failing tests I want to investigate but I'm pretty sure they're not related to your changes.
